File: lib/java/pplib/JsonParser.java
package pplib;

import java.io.IOException;
import java.util.List;

import org.json.simple.*;
import org.json.simple.parser.*;

import pplib.EasyFileReader;
import pplib.exceptions.*;

public class JsonParser{
	JSONParser parser;

	public JsonParser(){
		parser = new JSONParser();
	}

	public JSONObject createObject(String filename) throws PPJsonException{
		try{
			String jsonString = EasyFileReader.readFileToString(filename);
			JSONParser parser = new JSONParser();
			JSONObject object = (JSONObject)parser.parse(jsonString);
			return object;
		}
		catch (IOException|ParseException e){
			throw new PPJsonException(e.getMessage());
		}
	}

	public Object getRaw(JSONObject object, String tag){
		return object.get(tag);
	}

	public JSONObject getObject(JSONObject object, String tag){
		JSONObject retObject = (JSONObject)object.get(tag);
		return retObject;
	}

	public JSONArray getArray(JSONObject object, String tag){
		JSONArray array = (JSONArray)object.get(tag);
		return array;
	}

	public Object getString(JSONObject object, String tag){
		String string = (String)object.get(tag);
		return string;
	}

	public int getInt(JSONObject object, String tag){
		int integer = ((Long)object.get(tag)).intValue();
		return integer;
	}
}

File: lib/python/pplib/json_parser.py
import json

class PPjo:
	def __init__(self,string=None,obj=None,filename=None):
		if string is not None:
			self.obj = json.loads(string)
		elif filename is not None:
			with open(filename) as f:
				self.obj = json.load(f)
		elif obj is not None:
			self.obj = obj

	def get_object(self,tag):
		return PPjo(obj=self.obj.get(tag))

	def get_string(self,tag):
		return str(self.obj.get(tag)).replace("u\'","\'").replace("\'","\\\"")

	def get_strings(self,tag):
		return list([str(i) for i in self.obj.get(tag)])

	def get_int(self,tag):
		return int(self.obj.get(tag))

	def get_ints(self,tag):
		return list([int(i) for i in self.obj.get(tag)])

	def get_bool(self,tag):
		return self.obj.get(tag) == 'true'

	def get_objects(self,tag):
		return list([PPjo(obj=obj) for obj in self.obj.get(tag)])

	def __repr__(self):
		return self.obj
